ci/no_atomic.sh

```diff
@@ -26,6 +26,31 @@ for target in $(rustc --print target-list); do
 done
 echo "];" >>"$file"
 
+{
+    # Only crossbeam-utils actually uses this const.
+    echo "#[allow(dead_code)]"
+    echo "const NO_ATOMIC_64: &[&str] = &["
+} >>"$file"
+for target in $(rustc --print target-list); do
+    res=$(rustc --print target-spec-json -Z unstable-options --target "$target" \
+        | jq -r "select(.\"max-atomic-width\" == 32)")
+    [[ -z "$res" ]] || echo "    \"$target\"," >>"$file"
+done
+# It is not clear exactly what `"max-atomic-width" == null` means, but they
+# actually seem to have the same max-atomic-width as the target-pointer-width.
+# The targets currently included in this group are "mipsel-sony-psp",
+# "thumbv4t-none-eabi", "thumbv6m-none-eabi", all of which are
+# `"target-pointer-width" == "32"`, so assuming them `"max-atomic-width" == 32`
+# for now.
+for target in $(rustc --print target-list); do
+    res=$(rustc --print target-spec-json -Z unstable-options --target "$target" \
+        | jq -r "select(.\"max-atomic-width\" == null)")
+    [[ -z "$res" ]] || echo "    \"$target\"," >>"$file"
+done
+echo "];" >>"$file"
+
+# There is no `"max-atomic-width" == 16` or `"max-atomic-width" == 8` targets.
```

Comment:
Both "avr-unknown-gnu-atmega328" and "msp430-none-elf", which are `"target-pointer-width" == "16"`, do not support atomic:
- https://github.com/rust-lang/rust/blob/1.52.1/compiler/rustc_target/src/spec/avr_gnu_base.rs
- https://github.com/rust-lang/rust/blob/1.52.1/compiler/rustc_target/src/spec/msp430_none_elf.rs